channeldb/witness_cache.go

```diff
@@ -63,25 +64,44 @@ type WitnessCache struct {
 	db *DB
 }
 
+// witnessEntry is a key-value struct that holds each key -> witness pair, used
+// when inserting records into the cache.
+type witnessEntry struct {
+	key     []byte
+	witness []byte
+}
+
 // NewWitnessCache returns a new instance of the witness cache.
 func (d *DB) NewWitnessCache() *WitnessCache {
 	return &WitnessCache{
 		db: d,
 	}
 }
 
+// AddSHA256Witnesses adds a batch of new SHA256 preimages into the witness
+// cache. This is an alias for AddWitnesses that uses Sha256HashWitness as the
+// preimages' witness type.
+func (w *WitnessCache) AddSHA256Witnesses(preimages ...lnhash.Preimage) error {
```

Comment:
I could only find a single witness type. Why don't we remove it alltogether? Looks like something was added prematurely that we may never need?

Reply:
remove the older interface you mean? we do need to keep the constants to access the witnesses on disk

Reply:
Yes, remove the old interface.

Reply:
Just remembered that we will have other witness types when we move to discrete log  challenges, though seems like that can also be exposed only via the targeted interfaces instead of the general one. Also allows us to make `WitnessType` private

Reply:
decided to leave the existing interface for now, though i'm not opposed re: removing it in the future. main reason is to keep the test asserting the two are identical, as i did discover a bug in the initial implementation as a result of writing the test